aspnetcore/fundamentals/app-state/samples/3.x/SessionSample/Startup3.cs

```diff
@@ -0,0 +1,58 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace SessionSample
+{
+    public class Startup3
+    {
+        public Startup3(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        #region snippet1
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddControllersWithViews()
+                .AddSessionStateTempDataProvider();
+            services.AddRazorPages()
+                .AddSessionStateTempDataProvider();
```

Comment:
It ends up as a no-op, so might not matter, but only one of the calls to `AddSessionStateTempDataProvider` is required here. Maybe that's just a quirk of how these methods were split out in 3.x...

Reply:
@serpent5  I added
Only one call to `AddSessionStateTempDataProvider` is required